test/PowerShellEditorServices.Test.E2E/Processes/ServerProcess.cs

```diff
@@ -0,0 +1,141 @@
+using System;
+using System.IO;
+using System.Reactive.Subjects;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+
+namespace PowerShellEditorServices.Test.E2E
+{
+    /// <summary>
+    ///     A <see cref="ServerProcess"/> is responsible for launching or attaching to a language server, providing access to its input and output streams, and tracking its lifetime.
+    /// </summary>
+    public abstract class ServerProcess
+        : IDisposable
+    {
+        private readonly ISubject<System.Reactive.Unit> _exitedSubject;
+        /// <summary>
+        ///     Create a new <see cref="ServerProcess"/>.
+        /// </summary>
+        /// <param name="loggerFactory">
+        ///     The factory for loggers used by the process and its components.
+        /// </param>
+        protected ServerProcess(ILoggerFactory loggerFactory)
+        {
+            if (loggerFactory == null)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=5596138)